src/Compilers/CSharp/Test/Emit2/Semantics/ExtensionTypeTests.cs

```diff
@@ -19447,5 +19767,1026 @@ public class Nested { }
+        Assert.Empty(model.GetMemberGroup(memberAccess)); // PROTOTYPE need to fix the semantic model
+    }
+
+    [ConditionalFact(typeof(CoreClrOnly))]
+    public void ExtensionIndexerAccess_InapplicableInstanceIndexer()
+    {
+        // We look at indexers in extension types if the property group on type has no applicable candidates
+        var source = """
+System.Console.Write(new C()[42]);
+
+class C
+{
+    public int this[string s] => throw null;
+}
+
+implicit extension E for C
+{
+    public int this[int i] => i;
+}
+""";
+        var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+        comp.VerifyDiagnostics();
+
+        // PROTOTYPE Execute when adding support for emitting non-static members
+        //CompileAndVerify(comp, expectedOutput: "42", verify: Verification.FailsPEVerify)
+        //    .VerifyDiagnostics();
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+        var memberAccess = GetSyntax<ElementAccessExpressionSyntax>(tree, "new C()[42]");
+        Assert.Equal("System.Int32 E.this[System.Int32 i] { get; }", model.GetSymbolInfo(memberAccess).Symbol.ToTestDisplayString());
+
+        Assert.Empty(model.GetMemberGroup(memberAccess)); // PROTOTYPE need to fix the semantic model
+    }
+
+    [Fact]
+    public void ExtensionIndexerAccess_BaseReceiver()
+    {
+        var source = """
+class Base { }
+
+class C : Base
+{
+    public void M()
+    {
+        System.Console.Write(base[42]);
+    }
+}
+
+implicit extension E for Base
+{
+    public int this[int i] => throw null;
+}
+""";
+        var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+        // PROTOTYPE confirm we don't want to add special rules for `base`
+        comp.VerifyEmitDiagnostics(
+            // (7,30): error CS0021: Cannot apply indexing with [] to an expression of type 'Base'
+            //         System.Console.Write(base[42]);
+            Diagnostic(ErrorCode.ERR_BadIndexLHS, "base[42]").WithArguments("Base").WithLocation(7, 30)
+            );
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+        var memberAccess = GetSyntax<ElementAccessExpressionSyntax>(tree, "base[42]");
+        Assert.Null(model.GetSymbolInfo(memberAccess).Symbol);
+        Assert.Empty(model.GetMemberGroup(memberAccess));
+    }
+
+    [Fact]
+    public void ExtensionIndexerAccess_AmbiguityWithExtensionOnBaseType()
+    {
+        var source = """
+System.Console.Write(new C()[42]);
+
+class Base { }
+
+class C : Base { }
+
+implicit extension E1 for Base
+{
+    public int this[int i] => throw null;
+}
+
+implicit extension E2 for C
+{
+    public int this[int i] => i;
+}
+""";
+        var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+        comp.VerifyEmitDiagnostics(
+            // (1,22): error CS0021: Cannot apply indexing with [] to an expression of type 'C'
```

Comment:
>// (1,22): error CS0021: Cannot apply indexing with [] to an expression of type 'C' [](http://example.com/codeflow?start=12&length=83)

Consider adding a PROTOTYPE comment, E2 might be a better choice because it extends a more specific type. I think we have rules around receiver types for regular extension methods.